tools/r2d2/custom.py

```diff
@@ -0,0 +1,148 @@
+import pandas as pd
+import sys
+import time
+
+prediction_value = None
+
+
+class R2D2:
+    MEGA = 10 ** 6
+    MEGA_STR = " " * MEGA
+
+    CMD_COL = "cmd"
+    ARG_COL = "arg"
+
+    def __init__(self):
+        self._mem_array = []
+        print("R2D2 __init__")
+
+    def _mem_size_mb(self):
+        return len(self._mem_array)
+
+    def _clear_memory(self):
+        self._mem_array = []
+
+    def _alloc_additional_memory(self, additional_mb):
+        for i in range(0, additional_mb):
+            try:
+                self._mem_array.append(R2D2.MEGA_STR + str(i))
+                print("Adding: {}".format(i))
+            except MemoryError:
+                print("No more memory here..")
+                break
+
+            i += 1
+
+    def alloc_memory(self, arg):
+        # Always taking the info to run from the first command
+        memory_mb = int(arg)
+        print("Found memory_mb in data: {}".format(memory_mb))
+        if memory_mb <= 0:
+            print("Clearing memory")
+            self._clear_memory()
+        elif memory_mb > 0:
+            additional_mem = memory_mb - self._mem_size_mb()
```

Comment:
So sending 2 times command:
memory,2000 allocates just 2000 and not 4000
Mention it in the readme